processors/filtering/base_filter.py

```diff
@@ -45,25 +46,74 @@ def process(self):
         # Filter posts
         matching_posts = self.filter_items()
 
+        # Keep track of whether we need to copy over annotations and/or annotation fields
+        annotation_fields = self.source_dataset.annotation_fields
+        num_annotations = 0 if not annotation_fields else self.source_dataset.num_annotations()
+        all_annotations = None
+
+        # If this source dataset has less than n annotations, just retrieve them all before iterating
```

Comment:
Since this requires retrieving and then re-inserting all annotations, maybe a direct database query is possible instead? E.g.

```
INSERT INTO annotations (dataset, field_id, item_id, timestamp, timestamp_created, label, type, options, value, author, author_original, by_processor, metadata, from_dataset)
OVERRIDING USER VALUE 
SELECT '[new_key]', field_id, item_id, timestamp, timestamp_created, label, type, options, value, author, author_original, by_processor, metadata, from_dataset FROM annotations 
WHERE dataset = '[old_key]'
```

This would eliminate the python roundtrip which would probably speed things up considerably for large datasets. You would need some extra queries to update the field_id afterwards, but no database fetches are required that way.